lib/ssl/src/ssl_handshake.erl

```diff
@@ -850,28 +847,28 @@ decode_handshake(_, ?NEXT_PROTOCOL, <<?BYTE(SelectedProtocolLength),
     #next_protocol{selected_protocol = SelectedProtocol};
 
 decode_handshake(Version, ?SERVER_HELLO, <<?BYTE(Major), ?BYTE(Minor), Random:32/binary,
-					    ?BYTE(SID_length), Session_ID:SID_length/binary,
-					    Cipher_suite:2/binary, ?BYTE(Comp_method)>>) ->
+                                           ?BYTE(SID_length), Session_ID:SID_length/binary,
+                                           Cipher_suite:2/binary, ?BYTE(CompMethod)>>) ->
+    ?NO_COMPRESSION = CompMethod,
```

Comment:
can't you pattern match in function clause? you want to avoid that?
here and below.

Reply:
Fixed